src/main/java/org/opentripplanner/routing/api/request/preference/VehicleParkingPreferences.java

```diff
@@ -0,0 +1,228 @@
+
+/**
+ * The parking preferences contain preferences for car and bicycle parking. These preferences
+ * include filtering, preference and realtime usage.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE.
+ */
+public final class VehicleParkingPreferences implements Serializable {
+
+  public static final VehicleParkingPreferences DEFAULT = new VehicleParkingPreferences();
+  private final Cost unpreferredVehicleParkingTagCost;
+  private final boolean useAvailabilityInformation;
+  private final VehicleParkingFilterRequest filter;
+  private final VehicleParkingFilterRequest preferred;
+  private final Duration parkTime;
+  private final Cost parkCost;
+
+  /** Create a new instance with default values. */
+  private VehicleParkingPreferences() {
+    this.unpreferredVehicleParkingTagCost = Cost.costOfMinutes(5);
+    this.useAvailabilityInformation = true;
+    this.filter = VehicleParkingFilterRequest.empty();
+    this.preferred = VehicleParkingFilterRequest.empty();
+    this.parkTime = Duration.ofMinutes(1);
+    this.parkCost = Cost.costOfMinutes(2);
+  }
+
+  private VehicleParkingPreferences(Builder builder) {
+    this.unpreferredVehicleParkingTagCost = builder.unpreferredVehicleParkingTagCost;
+    this.useAvailabilityInformation = builder.useAvailabilityInformation;
+    this.filter =
+      new VehicleParkingFilterRequest(
+        builder.bannedVehicleParkingTags,
+        builder.requiredVehicleParkingTags
+      );
+    this.preferred =
+      new VehicleParkingFilterRequest(
+        builder.notPreferredVehicleParkingTags,
+        builder.preferredVehicleParkingTags
+      );
+    this.parkTime = builder.parkTime;
+    this.parkCost = builder.parkCost;
+  }
+
+  public static VehicleParkingPreferences.Builder of() {
+    return new Builder(DEFAULT);
+  }
+
+  /**
+   * What cost is applied to using parking that is not preferred.
+   */
+  public Cost unpreferredVehicleParkingTagCost() {
+    return unpreferredVehicleParkingTagCost;
+  }
+
+  /**
+   * Should availability information be used during routing.
+   */
+  public boolean useAvailabilityInformation() {
+    return useAvailabilityInformation;
+  }
+
+  /**
+   * Parking containing select filters must only be usable and parking containing with not filters
+   * cannot be used.
+   */
+  public VehicleParkingFilterRequest filter() {
+    return filter;
+  }
+
+  /**
+   * Which vehicle parking tags are preferred. Vehicle parking facilities that don't have one of these
+   * tags receive an extra cost.
+   * <p>
+   * This is useful if you want to use certain kind of facilities, like lockers for expensive e-bikes.
+   */
+  public VehicleParkingFilterRequest preferred() {
+    return preferred;
+  }
+
+  /** Time to park a vehicle */
+  public Duration parkTime() {
+    return parkTime;
+  }
+
+  /** Cost of parking a bike. */
+  public Cost parkCost() {
+    return parkCost;
+  }
```

Comment:
Should we expose (or even use) Cost and Duration?

Reply:
The preferences is part of the API - the purpose is to make it easy to use and safe - avoid mistakes. So, indeed we should use the type-safe wrapper types. The API is not about the internal model. When we do routing in AStar and Raptor we should copy/map the values into the AStar Request(does not exist) and into the Raptor Request. The Raptor and AStar requests should be consered with performance. In this way we make a nice user friendly API witch should hide the implementation details from the APIs. We can reuse types and structures - the mapping into Raptor and AStar can use the same source (walkReluctance, boardingCost ...). 

Short: Use the typesafe version of Cost and Reluctance here.

Reply:
I agree. Whatever call site needs something more efficient needs to create their own model.